commit-status-publisher-server/src/main/java/jetbrains/buildServer/commitPublisher/stash/StashPublisher.java

```diff
@@ -214,4 +229,58 @@ private String getUsername() {
   private String getPassword() {
     return myParams.get(Constants.STASH_PASSWORD);
   }
+
+  private StashCommit findPullRequestCommit(@NotNull SBuild build) throws PublisherException {
+    String apiUrl = myParams.get(Constants.STASH_BASE_URL);
+    String projectKey = myParams.get(Constants.STASH_PROJECT_KEY);
+    String repository = myParams.get(Constants.STASH_REPO_NAME);
+
+    final List<StashCommit> lastCommit = new ArrayList<StashCommit>();
+
+    // Plugin will recognize pull request only when Branch specification parameter will be configured as below:
+    // +:refs/pull-requests/(*/merge)
+    // https://blog.jetbrains.com/teamcity/2013/02/automatically-building-pull-requests-from-github-with-teamcity
+
+    if (build.getBranch().getName().contains("merge")) { // try to find pull request only for branches with merge in the name
+      String pullRequestNumber = build.getBranch().getName().replace("/merge",""); //remove merge from brache name to get only pull request number
+      apiUrl = apiUrl + "/rest/api/1.0/projects/" + projectKey + "/repos/" + repository + "/pull-requests/" + pullRequestNumber + "/commits/";
+      LOG.debug("Bitbucket Pull Request apiUrl: " + apiUrl);
+    }
+
+    if (null != apiUrl || apiUrl.length() != 0) {
+      try {
+        HttpResponseProcessor processor = new DefaultHttpResponseProcessor() {
+          @Override
+          public void processResponse(HttpResponse response) throws HttpPublisherException, IOException {
+
+            final HttpEntity entity = response.getEntity();
+            if (null == entity) {
+              throw new HttpPublisherException("Bitbucket publisher has received no response");
+            }
+            ByteArrayOutputStream bos = new ByteArrayOutputStream();
+            entity.writeTo(bos);
+            final String json = bos.toString("utf-8");
+            PullRequestInfo commitInfo = myGson.fromJson(json, PullRequestInfo.class);
```

Comment:
It seems if there is no PullRequestInfo in the response (e.g. the branch name contained "merge" substring for some other reason), the status publishing will fail. Is that right?